src/Columns/ColumnUnique.h

```diff
@@ -316,8 +316,8 @@ size_t ColumnUnique<ColumnType>::getNullValueIndex() const
     return 0;
 }
 
-
-namespace
```

Comment:
This change looks wrong (won't compile).

Reply:
I did compile locally using Clang 14 (Ubuntu 22.04 LTS, x86-64). What compiler were you using?

Reply:
In this case the rule is already enabled (or rather, not disabled) in the `.clang-tidy` configuration file of the project as far as I can tell:
```
clang-tidy-14 -p build src/Columns/ColumnUnique.h
178884 warnings generated.
/root/projects/github/ClibMouse/ClickHouse/src/Columns/ColumnUnique.h:320:1: error: do not use unnamed namespaces in header files [cert-dcl59-cpp,google-build-namespaces,-warnings-as-errors]
namespace
^
Suppressed 179630 warnings (178882 in non-user code, 748 NOLINT).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
1 warning treated as error
```

As for the solution itself, I merely reduced the scope of the `FieldVisitorGetData` class to the method that uses it exclusively. It may seem a bit odd, but it's actually syntactically correct. I believe that the original intent with the anonymous namespace was similar, but having it in a header file actually means that all translation units that include this header will have access to it.

Running the same check after making the change:
```
clang-tidy-14 -p build src/Columns/ColumnUnique.h
178882 warnings generated.
Suppressed 179630 warnings (178882 in non-user code, 748 NOLINT).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
```

Would you accept this particular modification in light of the explanation above?

Reply:
The [clang-tidy rule](https://clang.llvm.org/extra/clang-tidy/checks/google/build-namespaces.html) (respectively its rule aliases) is already enabled, or to be precise: not disabled. But it was not reported in CI, I think the reason is that clang-tidy findings in (the vast majority of) headers are suppressed, cf. #40741. However, when clang-tidy is called directly (like OP did) and I assume that ClickHouse's .clang-tidy file was not used, the issue pops up.

The fix lgtm. An alternative solution could have been to place the visitor into a names "detail" or "impl" namespace.